### PR TITLE
Bluespace rule + Events cleanup and improvements

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Reworked events code to support:
Warning before events ends (Usefull for bluespace events)
Radio channel support other then just using global

Reworked bluespace events code to support:
Multi grids spawn in the same event, min max.
No more FTL into other grids
Range min + max support per events
Components adding to grids
Localized naming for grids / random naming
Support to make an event grid stay and not get removed on event end
Support for spawning dungeons as events (Not yet added to events)

## Why / Balance
Allow the events to give us much more.

## How to test
addgame rule events.

## Media
N/A

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
Forks will need to update bluespace events to new format.

**Changelog**
:cl:
- add: Bluespace events now show a warning five minutes before ending.
- add: The broken McDelivery event now spawns in a set of grids with fewer food crates each.

### CODE REVIEW

File: Resources/Prototypes/_NF/Events/nf_events_bluespace_grids.yml [hunk 1]
```diff
@@ -72,14 +102,24 @@
...
+    groups:
+      grid: !type:GridSpawnGroup
+        minimumDistance: 1500
+        maximumDistance: 2500
+        addComponents:
+        - type: Gravity
+          enabled: true
+          inherent: true
+        - type: IFF
+          color: "#E10F9B"
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Events/nf_events_bluespace_grids.yml:120]: Should also add comp to make the grid static (not in air) or heavy
Should also make sure it cannot have its mass fucked with by shuttle computers

**Comment by dvir001** [comment]
This PR is ready and contain a major rework for bluespace events.

**Review by whatston3 — APPROVED** [review]
Seems to work fine.  Event selection with multiple grids works alright - all grids in a set are chosen in random order.

Left some really warning announcements (and changed the asteroid ending announcement to match the idea of inherent, unstable bluespace activity), but the timeout on them works well under test.

File: Resources/Prototypes/_NF/Events/nf_events_bluespace_grids.yml (hunk 1 above)
Comment (whatston3) [reply Resources/Prototypes/_NF/Events/nf_events_bluespace_grids.yml]: The latter part is fine, they aren't stations, so this shouldn't be an issue.

**Comment by dvir001** [comment]
Tested again, good to go.

**Comment by whatston3** [comment]
Seems fair enough, let's go.

**Pull request merged**